test/test_sphere.jl

```diff
@@ -51,6 +114,38 @@ function ManifoldsBase.project!(::TestSphere, Y, p, X)
     return Y
 end
 
+function Random.rand!(M::TestSphere, pX; vector_at = nothing, σ = one(eltype(pX)))
+    if vector_at === nothing
+        project!(M, pX, randn(eltype(pX), representation_size(M)))
+    else
+        n = σ * randn(eltype(pX), size(pX)) # Gaussian in embedding
+        project!(M, pX, vector_at, n) #project to TpM (keeps Gaussianness)
+    end
+    return pX
+end
```

Comment:
We've recently had a discussion about not needing the `rng`-free rand but I don't remember what was the conclusion. It seems to be needed now though?

Reply:
I think we o have the fallback from rig-free to rng-included ones – I feel a bit like exp/exp! here. A beginner might implement exp (or the rng-free rand!) the advanced the other two and we should always do the second (and half beginners boost their code?). So I would switch this?

Reply:
What do you mean by switching this?

Reply:
Implement the rng-version here ;)

Reply:
Both are needed actually.

Reply:
Oh, O had hoped we reduced that to one of the versions to reduce duplicate code.

Reply:
We did not, that would require more work. Ref. https://github.com/JuliaManifolds/Manifolds.jl/issues/652 .

Reply:
Then I remembered that wrongly (sorry, for today I am entering braindead mode by now, but teaching for the next project is nearly prepared ;) )

Reply:
No problem :slightly_smiling_face: . I guess we will just accept this duplication for now.